src/server/services/validate-upload.js

```diff
@@ -31,9 +31,10 @@ async function validateAgencyId ({ upload, records, trns }) {
     )
   }
 
-  // set agency id on the upload, for disambiguation
+  // always set agency id if possible; we omit passing the transaction in this
+  // case, so that the agency id gets set even if the upload fails to validate
   if (matchingAgency.id !== upload.agency_id) {
-    await setAgencyId(upload.id, matchingAgency.id, trns)
+    await setAgencyId(upload.id, matchingAgency.id)
```

Comment:
Just double-checking.  If the validation errors immediately above are triggered this should not run, correct?

Reply:
yes, those immediately `return` with the error. we only get here if there was no error.